### PR TITLE
0.10.0: MultiEnv/MultiMix, LinesRewards

### PR DESCRIPTION
**Pull request by Tezirg**
- Renames `MultiEnv` to `BaseMultiProcessEnv`
 - Create a couple implementations: `SingleEnvMultiProcess` for backwards compatibility, `MultiEnvMultiProcess` for new use case
 - Fixes some rewards ranges/computation

### CODE REVIEW

File: grid2op/Environment/MultiEnv.py [hunk 1]
```diff
@@ -11,383 +11,43 @@
...
-        """
-        Get the parameters of each sub environments
-        """
-        for remote in self._remotes:
-            remote.send(('params', None))
-        res = [remote.recv() for remote in self._remotes]
-        return res
-
+    def __init__(self, env, nb_env):
+        envs = [env for _ in range(nb_env)]
```
Comment (BDonnot) [inline grid2op/Environment/MultiEnv.py]: Are we sure it's working as intended ? Or do we need to perform copy of the env argument ?

File: grid2op/Environment/MultiEnv2.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    """
+    This class allows to evaluate a single agent instance on multiple environments running in parrallel.
+
+    Attributes
+    -----------
+    envs: `list:grid2op.Environment.Environment`
+        Al list of environments for which the evaluation will be made in parallel.
+
+    nb_envs: ``list:int``
+        Number of parallel underlying environment that will be handled.
```
Comment (BDonnot) [inline grid2op/Environment/MultiEnv2.py]: specify that the list should have the same size as the "envs" list and that total number of environments will be exactly the sum(nb_envs) maybe

File: grid2op/Environment/MultiEnv2.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+from grid2op.dtypes import dt_int
+from grid2op.Exceptions import Grid2OpException, MultiEnvException
+from grid2op.Space import GridObjects
+from grid2op.Environment import BaseMultiEnvironment
+from grid2op.Action import BaseAction
+
+# TODO test this class.
+
+class MultiEnvironment2(BaseMultiEnvironment):
```
Comment (BDonnot) [inline grid2op/Environment/MultiEnv2.py]: I don't like this name :-(

File: grid2op/Reward/LinesCapacityReward.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# If a copy of the Mozilla Public License, version 2.0 was not distributed with this file,
+# you can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+# This file is part of Grid2Op, Grid2Op a testbed platform to model sequential decision making in power systems.
+
+import numpy as np
+from grid2op.Reward.BaseReward import BaseReward
+from grid2op.dtypes import dt_float
+
+class LinesCapacityReward(BaseReward):
```
Comment (BDonnot) [inline grid2op/Reward/LinesCapacityReward.py:13]: what is the difference between that and l2rpn rewards ? Why having 2 rewards that does apparently the same thing ?

**Review by BDonnot — CHANGES_REQUESTED** [review]

File: grid2op/Environment/MultiEnv.py (hunk 1 above)
Comment (Tezirg) [reply grid2op/Environment/MultiEnv.py]: Yes

File: grid2op/Environment/MultiEnv2.py (hunk 3 above)
Comment (Tezirg) [reply grid2op/Environment/MultiEnv2.py]: Me neither

File: grid2op/Reward/LinesCapacityReward.py (hunk 4 above)
Comment (Tezirg) [reply grid2op/Reward/LinesCapacityReward.py:13]: This is linear and takes into account disconnected lines. I dont mind `rm -rf` the file if thats considered a duplicate

**Review by BDonnot — APPROVED** [review]

**Review by BDonnot — APPROVED** [review]

**Pull request merged**